core/app/models/spree/payment.rb

```diff
@@ -101,7 +109,10 @@ def offsets_total
     # @return [BigDecimal] the amount of this payment minus the offsets
     #   (old-style refunds) and refunds
     def credit_allowed
-      amount - (offsets_total.abs + refunds.sum(:amount))
+      amount - (
+        (Spree::Deprecation.silence { offsets_total.abs }) +
```

Comment:
What if we avoid the deprecation silencing and use a more verbose `self.class.where("source_type = 'Spree::Payment' AND amount < 0 AND state = 'completed' AND source_id = '#{self.id}'").pluck(:amount).sum`, which should do the same?